velox/functions/prestosql/tests/ArraySortTest.cpp

```diff
@@ -431,6 +431,32 @@ TEST_P(ArraySortTest, dictionaryEncodedElements) {
           {{4, std::nullopt}, {1, std::nullopt, std::nullopt}}));
 }
 
+TEST_F(ArraySortTest, complexArrayDecode) {
+  // Tests to see whether we handle decoding in cases array is composed of
+  // nested complex types.
+  auto arrayVector =
+      makeArrayVector<int64_t>({{1, 2}, {0, -1}, {3, 4}, {5, 6}});
+  auto dictionaryVector = BaseVector::wrapInDictionary(
+      makeNulls(8, nullEvery(3)),
+      makeIndices(8, [](auto row) { return (row + 1) % 4; }),
+      8,
+      arrayVector);
+  auto arrayOfArrays = makeArrayVector({0, 2, 4, 6}, dictionaryVector);
+  auto result = evaluate("array_sort(c0)", makeRowVector({arrayOfArrays}));
+
+  auto baseExpected = makeNullableArrayVector<int64_t>(
+      {{{5, 6}},
+       std::nullopt,
+       std::nullopt,
+       std::nullopt,
+       {{3, 4}},
+       {{5, 6}},
+       std::nullopt,
+       std::nullopt});
+  auto expected = makeArrayVector({0, 2, 4, 6}, baseExpected);
```

Comment:
```
nit: s/expected/expectedResult/
```

Reply:
Again trying to be consistent with other tests ; going to keep name expected unless you feel strongly enough.